service/frontend/domainHandler.go

```diff
@@ -613,6 +639,19 @@ func (d *domainHandlerImpl) createResponse(
 	return infoResult, configResult, replicationConfigResult
 }
 
+func (d *domainHandlerImpl) mergeBadBinaries(old map[string]*shared.BadBinaryInfo, new map[string]*shared.BadBinaryInfo) shared.BadBinaries {
```

Comment:
Would be good to have unit tests for such functions, should be simple to test

Reply:
yes, will do it.